Doc/c-api/decimal.rst

```diff
@@ -0,0 +1,231 @@
+.. sectionauthor:: Stefan Krah
+
+.. highlight:: c
+
+
+Decimal capsule API
+===================
+
+Capsule API functions can be used in the same manner as regular library
+functions, provided that the API has been initialized.
+
+
+Initialize
+----------
+
+Typically, a C extension module that uses the decimal API will do these
+steps in its init function:
+
+.. code-block::
+
+    #include <decimal.h>
```

Comment:
Is "decimal.h" really reachable at the toplevel of the include path? If so, isn't there a risk of name clash?

Reply:
Hmm, I copied what `datetime.h` is doing: It is in `Include/datetime.h` but not included in `Python.h`.  It is copied into e.g. `$prefix/include/python3.8m/datetime.h`, so I suppose distutils finds that.

It should probably be `#include "decimal.h"`, I'll check for a real  external extension module.

Reply:
The name `decimal.h` can certainly clash, but so can `datetime.h`.

Reply:
Indeed, I have no strong feelings about it. That said, it looks like there are [a number of projects](https://github.com/search?l=&q=filename%3Adecimal.h&type=Code) with include files named `decimal.h`, and there are already [complaints](https://bugreports.qt.io/browse/QTBUG-29985) about existing conflicts.

Perhaps naming it e.g. `pydecimal.h` would help avoid any ambiguities.

Reply:
Yes, that's probably safer.  I've renamed it to `pydecimal.h`, the instructions work with an installed `Python`.